### PR TITLE
[Unity] Add R.quantize and R.dequantize ops

### PR DESCRIPTION
**Pull request by ibsidorenko**
This commit adds 2 new operations (`R.quantize` and `R.dequantize`) and supports them in `LegalizeOps` pass.

### CODE REVIEW

**Comment by ibsidorenko** [comment]
cc @masahi @quic-sanirudh

File: python/tvm/relax/op/qdq.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    Parameters
+    ----------
+    data : tvm.relax.Expr
+        The input tensor to be dequantized.
+
+    scale : tvm.relax.Expr
+        The output scale.
+
+    zero_point : tvm.relay.Expr
+        The output zero_point.
```
Comment (masahi) [inline python/tvm/relax/op/qdq.py:74]: I think "input" scale / zp would be more correct?

File: src/relax/op/tensor/qdq.cc [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  // Check datatype of scale param:
+  if (scale_sinfo->dtype != DataType::Float(32) && scale_sinfo->dtype != DataType::Float(16)) {
+    ctx->ReportFatal(Diagnostic::Error(call)
+                     << "scale param datatype should be one of [float16, float32], but got "
+                     << scale_sinfo->dtype);
+  }
+
+  // Check datatype of zero_point param:
+  if (zp_sinfo->dtype != DataType::Int(32)) {
```
Comment (masahi) [inline src/relax/op/tensor/qdq.cc:76]: I remember that zp needs to be 8 bit in practice. How about enforcing int8 here?

File: src/relax/op/tensor/qdq.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  Sex The License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file src/relax/op/tensor/qdq.h
+ * \brief The functions to make Relax quantize/dequantize operator calls.
```
Comment (masahi) [inline src/relax/op/tensor/qdq.h:22]: Is this header needed?

**Review by masahi — COMMENTED** [review]
Also needs to fix the conflict.

File: src/relax/op/tensor/qdq.cc (hunk 2 above)
Comment (ibsidorenko) [reply src/relax/op/tensor/qdq.cc:76]: I think it is possible. I have never met the cases when `zero_point` is out of [-128, 127] range.

File: src/relax/op/tensor/qdq.h (hunk 3 above)
Comment (ibsidorenko) [reply src/relax/op/tensor/qdq.h:22]: There's no need for that in this PR. But it was added for the future when we want to construct `quantize`/`dequantize` op in one of TVM passes.

File: python/tvm/relax/op/qdq.py (hunk 1 above)
Comment (ibsidorenko) [reply python/tvm/relax/op/qdq.py]: Done

**Review by masahi — APPROVED** [review]

**Pull request merged**